### PR TITLE
Use Puppet-Datatype Sensitive for $masterauth and $auth_pass

### PR DESCRIPTION
**Pull request by cocker-cc**

### CODE REVIEW

**Review by kenyon — APPROVED** [review]

**Comment by ekohl** [comment]
Looks like there's a merge conflict now. Would you mind rebasing?

File: manifests/init.pp [hunk 1]
```diff
@@ -331,6 +331,12 @@
   Integer[0] $cluster_migration_barrier                          = 1,
   Hash[String[1], Hash] $instances                               = {},
 ) inherits redis::params {
+  $masterauth_unsensitive = if $masterauth =~ Sensitive {
+    $masterauth.unwrap
+  } else {
+    $masterauth
+  }
```
Comment (ekohl) [inline manifests/init.pp:338]: Where is this used?

**Review by ekohl — COMMENTED** [review]
Should we also set `show_diff` on the file resource if the value was passed in as `Sensitive`? I think something like this should work:
```puppet
file { '...':
  show_diff => $var !~ Sensitive,
}
```

File: manifests/init.pp (hunk 1 above)
Comment (cocker-cc) [reply manifests/init.pp:338]: Good Question.  It's used only in `manifests/instance.pp`, but not in `manifests/init.pp`.  I will remove these Lines.

**Comment by cocker-cc** [comment]
> Should we also set `show_diff` on the file resource if the value was passed in as `Sensitive`? I think something like this should work:
> 
> ```puppet
> file { '...':
>   show_diff => $var !~ Sensitive,
> }
> ```

With EPP instead of ERB the  `file`-Resource would behave automatically.  I created [this PR](https://github.com/voxpupuli/puppet-redis/pull/407) for therefor.

**Comment by ekohl** [comment]
Needs a rebase now that that PR is merged.

**Comment by cocker-cc** [comment]
> rebase

Done.

File: templates/redis.conf.epp [hunk 2]
```diff
@@ -1,74 +1,74 @@
...
+  Optional[String[1]]                                $syslog_facility,
+  Integer[1]                                         $databases,
+  Boolean                                            $save_db_to_disk,
+  Hash                                               $save_db_to_disk_interval,
+  Boolean                                            $stop_writes_on_bgsave_error,
+  Boolean                                            $rdbcompression,
+  Variant[String[1], Boolean]                        $dbfilename,
+  Stdlib::Absolutepath                               $workdir,
+  Optional[String[1]]                                $slaveof,
+  Optional[Variant[String[1], Sensitive[String[1]]]] $masterauth,
```
Comment (ekohl) [inline templates/redis.conf.epp:24]: I think you're missing the unwrap for this that got lose in a rebase.

Comment (cocker-cc) [reply templates/redis.conf.epp:24]: No, in EPP you don't need `unwrap`.  Thats the Point.

**Review by ekohl — APPROVED** [review]

**Comment by ekohl** [comment]
Thanks!

**Pull request merged**